app/Atro/Listeners/Entity.php

```diff
@@ -28,6 +28,17 @@ public function beforeSave(Event $event): void
                 ->createHook(Common\NextNumber::class)
                 ->beforeSave($event->getArgument('entity'), $event->getArgument('options'));
         }
+
```

Comment:
You have to find a better place to inject. Probably in the base repository. Where we do our validations.